woltka/file.py

```diff
@@ -156,15 +156,16 @@ def id2file_from_dir(dir_, ext=None, ids=None):
     """
     res = {}
     for fname in listdir(dir_):
-        try:
-            id_ = file2stem(fname, ext)
-        except ValueError:
-            continue
-        if ids and id_ not in ids:
-            continue
-        if id_ in res:
-            raise ValueError(f'Ambiguous files for ID: "{id_}".')
-        res[id_] = fname
+        if isfile(join(dir_, fname)):
```

Comment:
I'm not sure what you're going to be using this function's outputs for, but there is still the chance for a "race condition" where the file at this path is deleted (either by the user or by the system) *after* this check is run. As long as wherever you use this ID-to-file map in a place that handles the error where the file doesn't exist anymore, you should be ok.

Reply:
Oh, upon rereading this and the test, it looks like this isn't checking for existence -- it's checking that this is a file, and not a folder. Yeah, there's still theoretically the race condition case where a file exists but is then replaced with a directory ... probably less likely to happen though ;)

Reply:
I added this because @adswafford followed up in the same issue #28 . He created a subdirectory to hold some unwanted files (e.g., empty alignments), but Woltka thought the subdirectory is an alignment file. Therefore I added this feature to skip directories and only take files.

I added a `Notes` to the docstring to indicate this mechanism.